packages/back-end/src/models/ArchetypeModel.ts

```diff
@@ -54,24 +55,32 @@ export function parseArchetypeString(list: string) {
 }
 
 export async function createArchetype(
-  user: CreateArchetypeProps
+  arch: CreateArchetypeProps
 ): Promise<ArchetypeInterface> {
-  const newUser = await ArchetypeModel.create({
-    ...user,
+  const newArch = await ArchetypeModel.create({
+    ...arch,
     id: uniqid("sam_"),
     dateCreated: new Date(),
     dateUpdated: new Date(),
   });
-  return toInterface(newUser);
+  return toInterface(newArch);
 }
 
 export async function getAllArchetypes(
   organization: string,
-  owner: string
+  owner: string,
+  project?: string
 ): Promise<ArchetypeInterface[]> {
-  const archetype: ArchetypeDocument[] = await ArchetypeModel.find({
-    organization,
-  });
+  const query: FilterQuery<ArchetypeDocument> = {
+    organization: organization,
+    $or: [{ projects: { $exists: false } }, { projects: { $eq: [] } }], // returns archetypes that are not associated with a project (all projects)
+  };
+  // if project is set, return archetypes that are either all projects
+  // or are associated with the specified project
+  if (project && query["$or"]) {
+    query["$or"].push({ projects: { $eq: project } });
```

Comment:
Does this work? Checking if an array field equals a string?

Reply:
Yes, I believe so